src/_gettsim/transfers/elterngeld.py

```diff
@@ -365,14 +274,14 @@ def elterngeld_anteil_eink_erlass(
 
     # Lower replacement rate if considered income is above a threshold
     elif (
-        elterngeld_eink_relev_m
+        elterngeld_nettoeinkommen_vorjahr_m
         > elterngeld_params["nettoeinkommen_stufen"]["upper_threshold"]
     ):
         # Replacement rate is only lowered up to a specific value
         out = max(
             elterngeld_params["faktor"]
             - (
-                elterngeld_eink_relev_m
+                elterngeld_nettoeinkommen_vorjahr_m
                 - elterngeld_params["nettoeinkommen_stufen"]["upper_threshold"]
             )
             / elterngeld_params["eink_schritt_korrektur"],
```

Comment:
Was it a conscious decision to smooth out the function here? Because the decrease should only happen every second Euro, resulting in [this ](https://de.wikipedia.org/wiki/Elterngeld_(Deutschland)#/media/Datei:Elterngeld.png) wonderful functional relationship. (Same applies to the functional form below 1000€)

Reply:
I'm not entirely sure if I'm missing your point, but `elterngeld_params["eink_schritt_korrektur"]` is 2 so dividing by it takes care of that I think?

Reply:
Is this captured by the rounding rules? When `rounding == False`, we want to have a smooth relationship, else the sawtooth pattern (:scream:) from the pic Jakob linked.

Reply:
Jakob was of course right, we had only implemented the smoothed version. We now also have the correct calculation in there.

Reply:
Very likely that I'm overlpooking something, but I think there are two things not correct here:
1. `elterngeld_params["prozent_korrektur"]` is missing for incomes larger than `upper_threshold`. I think it should be 

```
        out = max(
            elterngeld_params["faktor"]
            - (
                elterngeld_nettoeinkommen_vorjahr_m
                - elterngeld_params["nettoeinkommen_stufen"]["upper_threshold"]
            )
            / elterngeld_params["eink_schritt_korrektur"] * elterngeld_params["prozent_korrektur"],
            elterngeld_params["prozent_minimum"],
        )
```


2. But even then it would not be exact, as the rounding is only to the next Euro if I understand `_nettoeinkommen_minus_obere_lohnersatzanteil_grenze` correctly. For example with an `elterngeld_nettoeinkommen_vorjahr_m` of 1201€, GETTSIM would caclulate a replacement rate of 0.67-(1201-1200)/2 * 0.001 = 0.6695. But the correct replacement rate would be 0.67 as the replacement rate discontinuously jumps to 0.669 at an income of 1202€ but is 0.667 at an income of 1201€.
So for the exact calculation of the replacement rate `elterngeld_nettoeinkommen_vorjahr_m` would be rounded downwards to the next multiple of 2. (I would love to know why it was designed like that)

Reply:
Regarding the second point: Rounding should be implemented via the usual rounding mechanic:

```python
rounding:
  _untere_lohnersatzanteil_grenze_minus_nettoeinkommen:
    1980-01-01:
      base: 2
      direction: down
      reference: §2 (2) BEEG
  _nettoeinkommen_minus_obere_lohnersatzanteil_grenze:
    1980-01-01:
      base: 2
      direction: down
      reference: §2 (2) BEEG
```

It is not hard-coded in the function to get a smoothed out version for applications in computational projects.

Regarding the first point: Yes, I think you're right. I didn't spend too much time on the formula, hoping it was implemented correctly in the past...

Reply:
Ah, that's so fancy!